src/Microsoft.AspNetCore.Sockets.Client.Http/HttpConnection.cs

```diff
@@ -412,7 +419,9 @@ private void SetTransferMode(TransferMode transferMode)
 
             if (Interlocked.Exchange(ref _connectionState, ConnectionState.Disconnected) == ConnectionState.Initial)
             {
-                // the connection was never started so there is nothing to clean up
+                // The connection was never started.
+                // Connection is now unusable, so cleanup resources.
+                _httpClient.Dispose();
```

Comment:
I think the client will not be disposed if DisposeAsync is called when the connection is in the connecting state - if this [condition](https://github.com/aspnet/SignalR/pull/999/files#diff-ffe775cce0a6518263da4dfc756b867eR157) is not true.